3/tests/unit/GraphQLBackend.spec.js

```diff
@@ -0,0 +1,312 @@
+            resolvers,
+            dataSources: () => ({
+                ds: new TodoAPI(tmpMock)
+            }),
+            mockEntireSchema: false,
+            formatError: (err) => {
+                console.log(err.stack);
+                return err
+            }
+        });
+
+        const {
+            query
+        } = createTestClient(server);
+
+        let testUser = createJwt("secret")
+
+        const addresult = await query({
+            query: ADD_TODO,
+            variables: { id: 10 , newMessage: "newentry", userAuth: testUser}
+        });
+
+        //console.log(addresult)
+        
+        const res = await query({
+            query: GET_TODO,
+            variables: {
+                id: 10
+            }
+        });
+        //console.log(res)
+        expect(res.data.todo.message).toEqual("newentry");
+    });
+
+    it('updates a todo', async() => {
+
+        var tmpMock = [{id: 1,message:"a", userAuth: "0"},{id: 2,message:"b", userAuth: "0"}]
+
+        const server = new ApolloServer({
+            typeDefs,
+            resolvers,
+            dataSources: () => ({
+                ds: new TodoAPI(tmpMock)
+            }),
+            mockEntireSchema: false,
+            formatError: (err) => {
+                console.log(err.stack);
+                return err
+            }
+        });
+
+        const {
+            query
+        } = createTestClient(server);
+
+        const updateresult = await query({
+            query: UPDATE_TODO,
+            variables: { id: 2 , updateMessage: "newmessage" }
+        });
+
+        //console.log(updateresult)
+
+        const res = await query({
+            query: GET_TODO,
+            variables: { id: 2}
+        });
+
+        
+        //console.log(res)
+        expect(res.data.todo.message).toEqual("newmessage");
+    });
+
+it('deletes a todo', async() => {
+
+    var tmpMock = MOCK_UP_DATASOURCE_CONTENT.slice(0)
+
+        const server = new ApolloServer({
+            typeDefs,
+            resolvers,
+            dataSources: () => ({
+                ds: new TodoAPI(tmpMock)
+            }),
+            mockEntireSchema: false,
+            formatError: (err) => {
+                console.log(err.stack);
+                return err
+            }
+        });
+
+        const {
+            query
+        } = createTestClient(server);
+
+        let testUser = createJwt("secret")
+
+        const addresult = await query({
+            query: ADD_TODO,
+            variables: { id: 6 , newMessage: "newentry", userAuth: testUser }
+        });
+
+        const delresult = await query({
+            query: DELETE_TODO,
+            variables: {
+                id: 6
+            }
+        });
+
+        const res = await query({
+            query: GET_TODOS,
+            variables: { }
+        });
+        //console.log(res)
+        expect(res.data.todos).toEqual(MOCK_UP_TEST_RESULTS);
+    });
+
+
+});
+
+describe("JWT tests", () => {
```

Comment:
```suggestion
describe("JWT", () => {
```
It's all tests anyways.